src/borg/remote.py

```diff
@@ -263,14 +263,26 @@ def negotiate(self, client_data):
         # not a known old format, send newest negotiate this version knows
         return {'server_version': BORG_VERSION}
 
-    def open(self, path, create=False, lock_wait=None, lock=True, exclusive=None, append_only=False):
+    def _resolve_path(self, path):
         if isinstance(path, bytes):
             path = os.fsdecode(path)
-        if path.startswith('/~'):  # /~/x = path x relative to home dir, /~username/x = relative to "user" home dir
-            path = os.path.join(get_home_dir(), path[2:])  # XXX check this (see also 1.0-maint), is it correct for ~u?
+        # Leading slash is always present with URI (ssh://), but not with short-form (who@host:path).
+        if path.startswith('/~/'):  # /~/x = path x relative to home dir
+            path = os.path.join(get_home_dir(), path[3:])
+        elif path.startswith('~/'):
+            path = os.path.join(get_home_dir(), path[2:])
```

Comment:
so, the stuff above is different from 1.0-maint now.

the stuff below corresponds to 1.0-maint.

is that what we want? is 1.0 thus incorrect somehow?